pointcloud/basic_octree_utils.go

```diff
@@ -23,20 +27,37 @@ func newInternalNode(tree []*BasicOctree) basicOctreeNode {
 		children: tree,
 		nodeType: internalNode,
 		point:    PointAndData{},
+		maxProb:  emptyProb,
 	}
 	return octNode
 }
 
 // Creates a new LeafNodeFilled and stores specified position and data.
 func newLeafNodeFilled(p r3.Vector, d Data) basicOctreeNode {
+	maxProb := getRawProb(d)
 	octNode := basicOctreeNode{
 		children: nil,
 		nodeType: leafNodeFilled,
 		point:    PointAndData{P: p, D: d},
+		maxProb:  maxProb,
 	}
 	return octNode
 }
 
+// getRawProb checks the data param for newLeafNodeFilled and returns a float.
```

Comment:
[nit[ make comment a bit more generic. 

// getRawProb returns the data param as a probability value